### PR TITLE
Check for unsynched commits before generating GH merge PR

### PR DESCRIPTION
**Pull request by JoeRobich**
Resolves https://github.com/dotnet/roslyn-tools/issues/1090

### CODE REVIEW

File: src/GithubMergeTool/GithubMergeTool.cs [hunk 1]
```diff
@@ -303,7 +328,8 @@ git commit
             }
 
             // https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge
-            response = await _client.PostAsyncAsJson("https://api.github.com/graphql", JsonConvert.SerializeObject(new {
+            response = await _client.PostAsyncAsJson("https://api.github.com/graphql", JsonConvert.SerializeObject(new
```
Comment (RikkiGibson) [inline src/GithubMergeTool/GithubMergeTool.cs:331]: nit: formatting change looks unnecessary/inconsistent.

Comment (JoeRobich) [reply src/GithubMergeTool/GithubMergeTool.cs:331]: Not sure if it is inconsistent.  Other uses of object literal notation in this code file we have a newline before the opening brace.

Comment (RikkiGibson) [reply src/GithubMergeTool/GithubMergeTool.cs:331]: I misread it, sorry 😓

**Comment by MiYanni** [comment]
Thanks for getting this change in! It reduces a lot of noise in PRs constantly having 'recent updates'.

**Pull request merged**